### ISSUE TITLE
"Error when processing your query"

### ISSUE THREAD

**Issue by karafecho** [NCATSTranslator/Feedback#528]
What drugs may treat fibrotic liver disease?

PK = 8355faa9-57df-4808-9dbb-4db6fcb30c3a (from prod)

![image](https://github.com/NCATSTranslator/Feedback/assets/26254388/162e0118-284e-4bd5-ab0c-4899a61ace3b)

**Comment by sandrine-m** [comment]
I have checked this [PK on ARAX](https://arax.ci.transltr.io/?r=8355faa9-57df-4808-9dbb-4db6fcb30c3a) and the query did run but returned no results from ARAs due to a lot of timeouts:
<img width="743" alt="image" src="https://github.com/NCATSTranslator/Feedback/assets/66342541/091876a3-2e61-417f-8d19-5b9b36cc526b">
Does the UI or ARS maintain a log of timeouts?

**Comment by sstemann** [comment]
it looks like its happening in Test:  https://ui.test.transltr.io/main/results?l=Fibrotic%20Liver%20Disease&i=MONDO:0100430&t=0&q=08421d36-2e71-4705-a389-98d55bbb11dd

PK: 08421d36-2e71-4705-a389-98d55bbb11dd

and CI:  https://ui.ci.transltr.io/main/results?l=Fibrotic%20Liver%20Disease&i=MONDO:0100430&t=0&q=70cdc8f1-4a57-4be0-a41e-495ff9e718dd

PK:  70cdc8f1-4a57-4be0-a41e-495ff9e718dd

@MarkDWilliams it looks like this might be specific to this query, given that in the ARAX UI it looks like other queries ran AFTER (my runs) in Test and CI this AM.  Anyway to confirm that?

**Comment by karafecho** [comment]
Thanks for looking into this issue, Sandrine and Sarah. To clarify, I ran the initial query that surfaced the issue in prod.

**Comment by andrewsu** [comment]
It looks like improving and unsecret are reproducibly timing out, and it looks like aragorn, arax, and BTE are reproducibly returning zero results.  Examining BTE in more detail, there doesn't appear to be anything going wrong -- it just is not finding any results.  I tried the same query for "liver disorder" (MONDO:0005154) (parent of "fibrotic liver disease") and "hepatorenal syndrome" (MONDO:0001382) (a sibling of "fibrotic liver disease") -- both of those queries return results just fine.  So there is no indication of any error in execution from the BTE side.  If the same is true for aragorn and arax, then I think this is a UI issue, which should report a "no results found" rather than a generic "error".

**Comment by edeutsch** [comment]
Agreed. I don't fully understand why there are 0 results for this MONDO, but it seems that ARAX is simply coming up with no results without obvious signs of malfunction.

Apparently our model has nothing here and all the other sources we check have nothing:
![image](https://github.com/NCATSTranslator/Feedback/assets/12707718/93d24836-76cc-4ce1-898c-3f8624317263)

**Comment by karafecho** [comment]
FYI (from # taqa-facilitators slack channel):

Sarah Stemann (Link Brokers)
  [2:25 PM](https://ncatstranslator.slack.com/archives/C05D7H990DV/p1694456759537209)
[@Sierra Moxon (SRI)](https://ncatstranslator.slack.com/team/U01JNQBNVEC)
 i think this MONDO is obsolete "0100430" that's the MONDO in the qGraph from "Fibrotic Liver Disease" and i can't find it here https://ols.monarchinitiative.org/search?q=0100430 but i do see it here:  https://www.ebi.ac.uk/ols/ontologies/mondo/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FMONDO_0100430
on OLS i can't free text search for it either - i get a little confused what version the name resolver is/should  be using?

Sierra Moxon (SRI)
  [2:33 PM](https://ncatstranslator.slack.com/archives/C05D7H990DV/p1694457213400889)
TLDR: too many software components trying to update :sob:
new monarch UI: https://next.monarchinitiative.org/node/MONDO:0100430#overview
new OLS UI: https://www.ebi.ac.uk/ols4/ontologies/mondo/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FMONDO_0100430
current version of MONDO (OWL): https://purl.obolibrary.org/obo/mondo.owl
current version of MONDO (obo): https://purl.obolibrary.org/obo/mondo.obo (edited) 
[2:34](https://ncatstranslator.slack.com/archives/C05D7H990DV/p1694457263325259)
Hopefully name resolver is pulling the ontology and its synonyms/aliases in from one of the purls above, but if they are pulling from OLS, they should be pulling from the OLS4 version of OLS.
[2:34](https://ncatstranslator.slack.com/archives/C05D7H990DV/p1694457275736839)
I don't think they can pull directly from monarch, but I am checking that.
[2:35](https://ncatstranslator.slack.com/archives/C05D7H990DV/p1694457342692089)
(old monarch, aka: current [monarchinitiative.org](http://monarchinitiative.org/) has an outdated version of MONDO - they are looking to release the new Monarch UI towards the end of September)

Sarah Stemann (Link Brokers)
  [2:44 PM](https://ncatstranslator.slack.com/archives/C05D7H990DV/p1694457898304679)
okay so it could have worked

Kara Fecho (Exposures Provider, SRI)
  [9:36 AM](https://ncatstranslator.slack.com/archives/C05D7H990DV/p1694525775568649)
So, is the conclusion that Name Resolver is pulling from an outdated version of MONDO? [@Gaurav Vaidya (SRI)](https://ncatstranslator.slack.com/team/UT28RAC2F)
: Any insights? (edited)

**Comment by sierra-moxon** [comment]
I think these two threads got sort of conflated; there is nothing wrong with the Mondo version used by Name resolver w/re to this issue.

**Comment by sstemann** [comment]
@brettasmi @webyrd could you all report back why your tools are reporting these errors on this query, which we're getting this error in all three environments, the other ARAs returned 0s:

Error 598 -
Rendering TRAPI 1.4 (presumed) message...no message!
Response contains no message, and hence no results.

**Comment by kaiwenho** [comment]
I have run in all three environments with the curl command against Unsecret server, and all return 0 answers without the timeout error. That means there is no answer found from Unsecret. 
However, every time Unsecret returns a 0 answer, we get a 598 timeout error, even though Unsecret server replies with the HTTP 200 OK success status. We have discussed the issue with Mark but we haven't found the solutions yet.

**Comment by brettasmi** [comment]
☝️ same with imProving.

Is it possible that this is an issue with the ARS?

**Comment by gprice1129** [comment]
@karafecho just re-tested this and it does not seem to occur anymore. Can we close this?

**Comment by karafecho** [comment]
Yup, closing now ...
